File: FantasyCXX/aura-vision/src/core/request/FaceRequest.cpp

#include "vision/core/request/FaceRequest.h"
#include "vision/config/runtime_config/RtConfig.h"

namespace aura::vision {

FaceRequest::FaceRequest(RtConfig* cfg)
    : AbsVisionRequest(),
      faceCount(static_cast<int>(cfg->faceMaxCount)), landmark2d106(nullptr) {
}

FaceRequest::~FaceRequest() { landmark2d106 = nullptr;
}

short FaceRequest::tag() const {
    return TAG;
}

void FaceRequest::clear() {
    AbsVisionRequest::clear();
    landmark2d106 = nullptr;
}

void FaceRequest::clearAll() {
    clear();
}

} // namespace aura::vision

File: FantasyCXX/aura-vision/android/vision_ability/src/main/java/com/baidu/iov/vision/ability/core/request/FaceRequest.java
package com.baidu.iov.vision.ability.core.request;

/**
 * 人脸相关请求基类
 * <p>
 * Created by liwendong on 2018/6/12.
 */
public class FaceRequest extends AbsVisionRequest {
    /**
     * 每帧的图像数据
     */
    public float[] mLandmarkData;

    @Override
    public void clear() {
        super.clear();
        mLandmarkData = null;
    }
}
